lib/chef/azure/chefhandlers/exception_handler.rb

```diff
@@ -30,14 +31,14 @@ def report
         load_azure_env
         message = "Check log file for details...\nBacktrace:\n"
         message << Array(backtrace).join("\n")
-        report_heart_beat_to_azure(AzureHeartBeat::READY, 0, "chef-service is running properly. Chef client run failed with error- #{message}")
+        report_heart_beat_to_azure(AzureHeartBeat::NOTREADY, 1, "chef-service is running properly. Chef client run failed with error- #{message}")

```

Comment:
At this point do we know that the `chef-service is running properly`? - Can we just report the Chef client run status here and remove the message about chef-service?

Reply:
I just updated the the status code to 1 since the block is about failure.   report_heart_beat_to_azure() This method is use for reporting hearbeat to Azure. And currently we are kept it off since as we discussed earlier we don't have mechanism to handle this. So I am just making sure the code has proper exit status and the state.